codersdk/workspaceagents_internal_test.go

```diff
@@ -50,7 +50,7 @@ func TestTailnetAPIConnector_Disconnects(t *testing.T) {
 		if !assert.NoError(t, err) {
 			return
 		}
-		ctx, nc := websocketNetConn(r.Context(), sws, websocket.MessageBinary)
+		ctx, nc := WebsocketNetConn(r.Context(), sws, websocket.MessageBinary)
```

Comment:
I'd prefer to fix this bad use of `r.Context()` since someone could look at this and get the wrong idea. (The `websocket.Accept` invalidates `r.Context()` cancellation.)

```suggestion
		ctx, nc := WebsocketNetConn(context.Background(), sws, websocket.MessageBinary)
```

I know some other places are using `r.Context()` via `ctx := r.Context()` too, but this one just seems too explicitly wrong. 😄 

Perhaps this is actually a case for moving `websocket.Accept` into `codersdk.WebsocketNetConn` as well (or creating a unified function, `codersdk.WebsocketAcceptNetConn`).

Reply:
> The websocket.Accept invalidates r.Context() cancellation.

I don't understand what it means to "invalidate" a context cancelation.

Are you referring to the idea that once we have a websocket, passing `r.Context()` is a bit pointless because the context won't get canceled before the underlying TCP connection is closed?  I guess that's true enough, but changing it to `context.Background()` doesn't change anything from a functional standpoint.

Are you suggesting we don't accept a context at all?

Reply:
The context is actually not even tied to the TCP connection, it is tied to the handler which creates a scenario where it’s unlikely to ever be cancelled.

https://pkg.go.dev/net/http#Hijacker

Functionally, you’re right, there’s no difference but this behavior can easily trip anyone up so its usage should be discouraged.